### PR TITLE
Add support for both cool and heat setpoint devices on a thermostat

### PR DESCRIPTION
**Pull request by bytedreamer**
The setpoint virtual device id will need to be changed to support both setpoints. I would assume this could cause issue if existing thermostat devices are configured.

### CODE REVIEW

**Comment by bytedreamer** [comment]
Pull request also includes a fix for Logical Rules

**Comment by bytedreamer** [comment]
Reverted fix to match upstream change

File: modules/ZWave/index.js [hunk 1]
```diff
@@ -1749,53 +1749,48 @@ ZWave.prototype.parseAddCommandClass = function (nodeId, instanceId, commandClas
...
-								icon: 'thermostat',
-								title: compileTitle("Thermostat", vDevIdNI)
-							}
-						},
-						overlay: {},
-						handler: function (command, args) {
-							instance.ThermostatSetPoint.Set(mode, args.level);
-
-							// and turn change the mode
-							var _v = this.controller.devices.get(deviceNamePrefix + self.CC["ThermostatMode"]);
```
Comment (PoltoS) [inline modules/ZWave/index.js]: This code automatically selected the desired mode if setpoint is change. I think it is worth to keep this.

File: modules/ZWave/index.js [hunk 2]
```diff
@@ -1749,53 +1749,48 @@ ZWave.prototype.parseAddCommandClass = function (nodeId, instanceId, commandClas
 				}
 			}
 
-			if (withTemp && !self.controller.devices.get(deviceNamePrefix + this.CC["ThermostatSetPoint"])) {
+			if (withTemp) {
 				var withTempHeat = instance.ThermostatSetPoint.data[MODE_HEAT],
```
Comment (PoltoS) [inline modules/ZWave/index.js]: The previous code worked perfectly for heat/cool if both are preset (setpoint is used for both modes, switch changes between cool and heat).

For off/heat and off/cool setpoint is used for temperature and switch to turn off/on.

In your case there would be two widgets for setpoint (while only one can be currently active) and how to switch between modes?

**Comment by bytedreamer** [comment]
I'm using the 2gig CT100 thermostat. Using the previous code, I couldn't set the cool temperature. Only the heat temperature is available. The 2gig thermostat has four modes of operation, off, heat, cool, and auto. If it is set to auto, both setpoints are used.

In my custom [application](﻿﻿https://github.com/bytedreamer/BusyBot), I allow the user to select either setpoint. It would default to the setpoint that is closest to the actual temperature. Without both setpoints available, it would be impossible to use the automation engine's API. For now, I need to use the Zway library. In the long term, I would like to switch over the automation engine.

Thanks

**Comment by PoltoS** [comment]
Interesting point! We don't use these HVAC in Europe, so I was always confused by this off/heat/cool/auto modes... In Europe we have heat only or off/heat thermostats.

There is a good reason to implemet like you say. We will think of it, thanks

**Comment by bytedreamer** [comment]
@PoltoS Any ideas to add new functionality without causing incompatibilities with the Europe's HVAC systems?

**Comment by PoltoS** [comment]
I think finally we will make a new widget for US HVAC

**Comment by bytedreamer** [comment]
Is there anything I can do to assist?

Thanks

**Comment by PoltoS** [comment]
Unmergeable, but fixed in another way

**Pull request closed without merging**